mula/tests/integration/test_services.py

```diff
@@ -0,0 +1,194 @@
+import copy
+import time
+import unittest
+from unittest import mock
+
+from scheduler import config, models, storage
+from scheduler.connectors import services
+from scheduler.utils import remove_trailing_slash
+
+from tests.factories import PluginFactory
+
+
+class BytesTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        self.config = config.settings.Settings()
+        self.service_bytes = services.Bytes(
+            host=remove_trailing_slash(str(self.config.host_bytes)),
+            user=self.config.host_bytes_user,
+            password=self.config.host_bytes_password,
+            source="scheduler_test",
+        )
+
+    def test_login(self):
+        self.service_bytes.login()
+
+        self.assertIsNotNone(self.service_bytes.headers)
+        self.assertIsNotNone(self.service_bytes.headers.get("Authorization"))
+
+    def test_expired_token_refresh(self):
+        self.service_bytes.get_last_run_boefje(
+            boefje_id="boefje-1",
+            input_ooi="ooi-1",
+            organization_id="org-1",
+        )
+        initial_token = copy.deepcopy(self.service_bytes.headers.get("Authorization"))
+
+        time.sleep(70)
+
+        self.service_bytes.get_last_run_boefje(
+            boefje_id="boefje-1",
+            input_ooi="ooi-1",
+            organization_id="org-1",
+        )
+        refresh_token = copy.deepcopy(self.service_bytes.headers.get("Authorization"))
+
+        self.assertNotEqual(initial_token, refresh_token)
+
+
+class KatalogusTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        self.config = config.settings.Settings()
+        self.dbconn = storage.DBConn(str(self.config.db_uri))
+
+        self.service_katalogus = services.Katalogus(
+            host=remove_trailing_slash(str(self.config.host_katalogus)),
+            source="scheduler_test",
+            cache_ttl=12345,
+        )
+
+    def tearDown(self) -> None:
+        self.service_katalogus.organisations_plugin_cache.reset()
+        self.service_katalogus.organisations_boefje_type_cache.reset()
+        self.service_katalogus.organisations_normalizer_type_cache.reset()
+        models.Base.metadata.drop_all(self.dbconn.engine)
+
+    @mock.patch("scheduler.connectors.services.Katalogus.get_organisations")
+    def test_flush_organisations_plugin_cache(self, mock_get_organisations):
+        # Mock
+        mock_get_organisations.return_value = [
+            models.Organisation(id="org-1", name="org-1"),
+            models.Organisation(id="org-2", name="org-2"),
+        ]
+
+        # Act
+        self.service_katalogus.flush_organisations_plugin_cache()
+
+        # Assert
+        self.assertEqual(len(self.service_katalogus.organisations_plugin_cache), 2)
+        self.assertIsNotNone(self.service_katalogus.organisations_plugin_cache.get("org-1"))
```

Comment:
I think it's better to test something like: `self.assertCountEqual(self.service_....cache.keys(), ("org-1", "org-2"))` (untested pseudocode)

This way you test the content and number of elements.